pkg/querier/frontend/frontend.go

```diff
@@ -32,6 +32,7 @@ import (
 const (
 	// StatusClientClosedRequest is the status code for when a client request cancellation of an http request
 	StatusClientClosedRequest = 499
+	maxBodySize               = 10 * 1024 * 1024 // 10 MiB
```

Comment:
Remember to also mention this as `[CHANGE]` in the `CHANGELOG` because after this PR gets merged we'll not allow anymore any query POST request with a body larger than 10MB.